### PR TITLE
Test Driven Development Project

### PR DESCRIPTION
**Pull request by BenjaminSemah**
### In this branch, we have done the following
✅ used the TDD approach to add these methods to the `Solver` class
- `factorial` method
- `reverse` method
- `fizzbuzz` method

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            expect(@solve.factorial(4)).to eql(24)
+            expect(@solve.factorial(5)).to eql(120)
+            expect(@solve.factorial(10)).to eql(3_628_800)
+            expect(@solve.factorial(2)).to eql(2)
+            expect(@solve.factorial(7)).to eql(5_040)
+        end
+
+        it 'checks that the factorial of 0 returns 1' do
+            expect(@solve.factorial(0)).to eql(1)
+        end
```
Comment (Kingstalux) [inline spec/solver_spec.rb]: - [x] Great job so far 👍 Kindly implement a test and a method for resolving the factorial of a negative number 💯

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @BenjaminSemah and @Yerimah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
149637855-7722d04c-b45d-4185-925f-7de047106461


## Highlights ⚡ 
✅ Good pull request description and commit messages 👍 
✅ All tests are passing 🔥 
✅ Good implementations of the various methods 👏 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @BenjaminSemah and @Yerimah 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Github inline comments feature is used properly.
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BenjaminSemah** [comment]
Thanks @emmanuellekamwa

**Pull request merged**